server/stream.go

```diff
@@ -3836,17 +3829,17 @@ func (mset *stream) processJetStreamMsg(subject, reply string, hdr, msg []byte,
 	if noInterest {
 		mset.lseq = store.SkipMsg()
 		mset.lmsgId = msgId
+		// If we have a msgId make sure to save.
+		if msgId != _EMPTY_ {
+			mset.storeMsgIdLocked(&ddentry{msgId, seq, ts})
+		}
 		mset.mu.Unlock()
 
 		if canRespond {
 			response = append(pubAck, strconv.FormatUint(mset.lseq, 10)...)
```

Comment:
Should have captured mset.lseq above actually!

Reply:
Not following this one..

Reply:
See the mset.lseq in FormatUint()? We should capture mset.lseq, say as lseq, under the mset lock above.